torch/_inductor/ir.py

```diff
@@ -5181,13 +5185,14 @@ def __repr__(self):
     @staticmethod
     def find_device(tensor_args, example_output):
         if tensor_args:
-            return tensor_args[0].get_device()
+            devices = [arg.get_device() for arg in tensor_args if arg.get_device()]
+            return devices[0]
         if isinstance(example_output, torch.Tensor):
             return example_output.device
         if isinstance(example_output, (list, tuple)):
-            devices = {FallbackKernel.find_device(None, x) for x in example_output}
+            device_set = {FallbackKernel.find_device(None, x) for x in example_output}
```

Comment:
why this change ? sets are non deterministic so we should avoid

Reply:
This is just a rename from devices -> device_set